proto/iamanager/v5/iamanager.proto

```diff
@@ -0,0 +1,258 @@
+message ErrorInfo {
+    int32 aos_code  = 1;
+    int32 exit_code = 2;
+    string message  = 3;
+}
+
+message APIVersion {
+    uint64 version = 1;
+}
+
+message GetCertRequest {
+    string type   = 1;
+    bytes issuer  = 2;
+    string serial = 3;
+}
+
+message GetCertResponse {
+    string type     = 1;
+    string cert_url = 2;
+    string key_url  = 3;
+}
+
+message SystemInfo {
+    string system_id  = 1;
+    string unit_model = 2;
+}
+
+message Subjects {
+    repeated string subjects = 1;
+}
+
+message PermissionsRequest {
+    string secret               = 1;
+    string functional_server_id = 2;
+}
+
+message InstanceIdent {
+    string service_id = 1;
+    string subject_id = 2;
+    uint64 instance   = 3;
+}
+
+message PermissionsResponse {
+    InstanceIdent instance  = 1;
+    Permissions permissions = 2;
+}
+
+message NodesID {
+    repeated string ids = 1;
+}
+
+message GetCertTypesRequest {
+    string node_id = 1;
+}
+
+message CertTypes {
+    repeated string types = 1;
+}
+
+message CreateKeyRequest {
+    string node_id  = 1;
+    string subject  = 2;
+    string type     = 3;
+    string password = 4;
+}
+
+message CreateKeyResponse {
+    string node_id       = 1;
+    string type          = 2;
+    string csr           = 3;
+    ErrorInfo error_info = 4;
+}
+
+message ApplyCertRequest {
+    string node_id = 1;
+    string type    = 2;
+    string cert    = 3;
+}
+
+message ApplyCertResponse {
+    string node_id       = 1;
+    string type          = 2;
+    string cert_url      = 3;
+    string serial        = 4;
+    ErrorInfo error_info = 5;
+}
+
+message Permissions {
+    map<string, string> permissions = 1;
+}
+
+message RegisterInstanceRequest {
+    InstanceIdent instance               = 1;
+    map<string, Permissions> permissions = 2;
+}
+
+message RegisterInstanceResponse {
+    string secret = 1;
+}
+
+message UnregisterInstanceRequest {
+    InstanceIdent instance = 1;
+}
+
+message GetNodeInfoRequest {
+    string node_id = 1;
+}
+
+message PauseNodeRequest {
+    string node_id = 1;
+}
+
+message ResumeNodeRequest {
+    string node_id = 1;
+}
+
+message PartitionInfo {
+    string name           = 1;
+    repeated string types = 2;
+    uint64 total_size     = 3;
+}
+
+message CPUInfo {
+    string model_name  = 1;
+    uint64 num_cores   = 2;
+    uint64 num_threads = 3;
+    string arch        = 4;
+    string arch_family = 5;
+}
+
+message NodeAttribute {
+    string name  = 1;
+    string value = 2;
+}
+
+message NodeInfo {
+    string id                         = 1;
+    string type                       = 2;
+    string name                       = 3;
+    string status                     = 4;
+    string os_type                    = 5;
+    repeated CPUInfo cpus             = 6;
+    float max_dmips                   = 7;
+    uint64 total_ram                  = 8;
+    repeated PartitionInfo partitions = 9;
+    repeated NodeAttribute attrs      = 10;
+}
+
+message StartProvisioningRequest {
```

Comment:
For all the messages put Response message just after corresponding request.

Reply:
Done